### PR TITLE
Bug 1211341 - Remove bluetooth APIv1 code from Settings, r=yzen

### PR DESCRIPTION
**Pull request by gasolin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e71df209385214f5d5b0c81c0adea619f8469df0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dbee531c7fb4a7bd3b83efb8dfa27ab675ae94ac

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=751e03ad16d5e9c53da815341cfa93851ed586d9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=87139751acf517b82660cc7e1bbb3209f5eb5300

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b191d4a18ff2ed4385e4e67d4ee3c0594d74a83

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3c8c09f8ec66544000008cabb99bcfce5d82fa53

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=41dc849725d17aca146e61e02aaef877870fc5f4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ad10ebc010c70921102d9eb2bf0f6283aaf2cd83

**Comment by gasolin** [comment]
Note: to remove bluetooth_helper in config/require.js

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=31fadfda615444b28f9d3744eeee874e3dbda111

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=52dee20870b1c24fb243959f38e0bab98167c657

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b4c470ec25b59b37d0ac78dc65f3f13c27185a2

File: apps/settings/test/unit/panels/about_more_info/hardware_info_test.js [hunk 1]
```diff
@@ -40,9 +35,6 @@ suite('about hardware info >', function() {
       observe: function() {},
       address: ''
     };
-    define('MockBluetooth1', function() {
-      return MockBluetooth1;
-    });
 
     MockBluetooth2 = {
```
Comment (yzen) [inline apps/settings/test/unit/panels/about_more_info/hardware_info_test.js]: Since we only have 1, lets rename it to MockBluetooth

Comment (yzen) [inline apps/settings/test/unit/panels/about_more_info/hardware_info_test.js]: Basically 2 in all names is a little misleading

File: apps/settings/js/panels/root/bluetooth_item.js [hunk 2]
```diff
@@ -152,14 +100,9 @@ define(function(require) {
...
     function bt__navigatePanelWithVersionCheck() {
-      if (this._APIVersion() === 1) {
-        // navigate old bluetooth panel..
-        SettingsService.navigate('bluetooth');
-      } else if (this._APIVersion() === 2) {
-        // navigate new bluetooth panel..
-        debug('navigate bluetooth_v2 panel');
-        SettingsService.navigate('bluetooth_v2');
-      }
+      // navigate new bluetooth panel..
```
Comment (yzen) [inline apps/settings/js/panels/root/bluetooth_item.js]: nit: navigate to bluetooth panel.

File: apps/settings/test/unit/panels/about_more_info/hardware_info_test.js (hunk 1 above)
Comment (gasolin) [inline apps/settings/test/unit/panels/about_more_info/hardware_info_test.js]: fixed

File: apps/settings/js/panels/root/bluetooth_item.js (hunk 2 above)
Comment (gasolin) [inline apps/settings/js/panels/root/bluetooth_item.js]: removed because the debug message below provide sufficient comment

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=106857ca115efe8bd6c597ea5f580445122ce1e8

**Pull request merged**